server/src/test/java/org/elasticsearch/cluster/coordination/CoordinatorTests.java

```diff
@@ -1713,7 +1713,18 @@ public void assertMatched() {
             Loggers.addAppender(joinHelperLogger, mockAppender);
             try {
                 cluster.runFor(
-                    defaultMillis(DISCOVERY_FIND_PEERS_INTERVAL_SETTING) + 2 * DEFAULT_DELAY_VARIABILITY,
+                    // This expects 8 tasks to be executed:
+                    //
+                    // * PeerFinder handling wakeup
+                    // * connectToRemoteMasterNode[0.0.0.0:11]
+                    // * [internal:transport/handshake] from {node1} to {node2}
+                    // * response to [internal:transport/handshake] from {node1} to {node2}
+                    // * [internal:discovery/request_peers] from {node1} to
+                    // * response to [internal:discovery/request_peers] from {node1} to {node2}
+                    // * [internal:cluster/coordination/join] from {node1} to {node2}
+                    // * error response to [internal:cluster/coordination/join] from {node1} to {node2}
```

Comment:
If we count the initial `PeerFinder handling wakeup` then there's actually 9 tasks in play:

```suggestion
                    // * PeerFinder handling wakeup
                    // * connectToRemoteMasterNode[0.0.0.0:11]
                    // * [internal:transport/handshake] from {node1} to {node2}
                    // * response to [internal:transport/handshake] from {node1} to {node2}
                    // * [internal:discovery/request_peers] from {node1} to
                    // * response to [internal:discovery/request_peers] from {node1} to {node2}
                    // * [internal:cluster/coordination/join] from {node1} to {node2}
                    // * [internal:transport/handshake] from {node2} to {node1} (rejected due to action block)
                    // * error response to [internal:cluster/coordination/join] from {node1} to {node2}
```

(the variability in the first task is captured by `defaultMillis()`, hence the `8`).